server/src/main/java/org/opensearch/search/startree/filter/DimensionFilterMerger.java

```diff
@@ -0,0 +1,208 @@
+                includeHigh = range1.isIncludeHigh();
+            } else if (comparison > 0) {
+                newHigh = high2;
+                includeHigh = range2.isIncludeHigh();
+            } else {
+                newHigh = high1;
+                includeHigh = range1.isIncludeHigh() && range2.isIncludeHigh();
+            }
+        }
+
+        // Check if range is valid
+        if (newLow != null && newHigh != null) {
+            int comparison = compareValues(newLow, newHigh);
+            if (comparison > 0 || (comparison == 0 && (!includeLow || !includeHigh))) {
+                return null; // No overlap
+            }
+        }
+
+        return new RangeMatchDimFilter(range1.getDimensionName(), newLow, newHigh, includeLow, includeHigh);
+    }
+
+    /**
+     * Intersects two exact match filters
+     * Returns null if no common values
+     */
+    private static DimensionFilter intersectExactMatchFilters(ExactMatchDimFilter exact1, ExactMatchDimFilter exact2) {
+        List<Object> values1 = exact1.getRawValues();
+        List<Object> values2 = exact2.getRawValues();
+
+        List<Object> intersection = new ArrayList<>();
+        for (Object value : values1) {
+            if (values2.contains(value)) {
+                intersection.add(value);
+            }
+        }
+
+        if (intersection.isEmpty()) {
+            return null;
+        }
+
+        return new ExactMatchDimFilter(exact1.getDimensionName(), intersection);
+    }
+
+    /**
+     * Intersects a range filter with an exact match filter.
+     * Returns null if no values from exact match are within range.
+     */
+    private static DimensionFilter intersectRangeWithExactMatch(RangeMatchDimFilter range, ExactMatchDimFilter exact) {
+        List<Object> validValues = new ArrayList<>();
+
+        for (Object value : exact.getRawValues()) {
+            if (isValueInRange(value, range)) {
+                validValues.add(value);
+            }
+        }
+
+        if (validValues.isEmpty()) {
+            return null;
+        }
+
+        return new ExactMatchDimFilter(exact.getDimensionName(), validValues);
+    }
+
+    /**
+     * Checks if a value falls within a range.
+     */
+    private static boolean isValueInRange(Object value, RangeMatchDimFilter range) {
+        if (range.getLow() != null) {
+            int comparison = compareValues(value, range.getLow());
+            if (comparison < 0 || (comparison == 0 && !range.isIncludeLow())) {
+                return false;
+            }
+        }
+
+        if (range.getHigh() != null) {
+            int comparison = compareValues(value, range.getHigh());
+            if (comparison > 0 || (comparison == 0 && !range.isIncludeHigh())) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Compares two values - Long for numeric fields and BytesRef for keywords field
+     */
+    private static int compareValues(Object v1, Object v2) {
```

Comment:
The field type related logic is spilling out into the class supposed to merge dimension filters.

Can we try out a way to push this towards DimensionFilterMapper instead ?

Reply:
Made the change. Can you please and let me know if the current changes looks good?

Reply:
Thanks @imRishN looks good. 

Do share your thoughts on https://github.com/opensearch-project/OpenSearch/pull/17941#discussion_r2053191386